sdl2-vga-terminal/src/VgaTerminal.hpp

```diff
@@ -56,50 +62,54 @@ class VgaTerminal : public Window
     void clear() noexcept;
   
     void moveCursorLeft() noexcept;
-    // TODO: double check moveCursorRight with incrementCursorPosition as a potential the same method.
-    //       moveCursorRight is the same as incrementalCursorPosition with disabled autoScroll. (?)
     void moveCursorRight() noexcept;
     void moveCursorUp() noexcept;
     void moveCursorDown() noexcept;
 
     void newLine() noexcept;
-    /// the X,Y are relative to the new viewport.
+
     bool setViewPort(const position_t& viewport, const uint8_t width, const uint8_t height) noexcept;
     bool setViewPort(const uint8_t x, const uint8_t y, const uint8_t width, const uint8_t height) noexcept;
     bool setViewPort(const SDL_Rect& r) noexcept;
     SDL_Rect getViewport() const noexcept;
     void resetViewport() noexcept;
+
+    const videoMode_t getMode() const noexcept;
     
-    uint8_t curDefaultCol = 7;
+    uint8_t cursorDefaultCol = 7;
+    uint16_t cursor_time = 500; /// ms
     bool showCursor = true;
     bool autoScroll = true;
-    
-    const videoMode_t getMode() const noexcept;
 private:
+    typedef struct _terminalChar_t : terminalChar_t
+    {
+        bool rendered;
+        bool operator==(const _terminalChar_t& o) const;
+    } _terminalChar_t;
+
     static const videoMode_t mode3;
     std::unique_ptr<SDL_Color[]> pCol;
-    SDL_Palette p;
+    SDL_Palette _pal;
     videoMode_t mode;
+    std::unique_ptr<_terminalChar_t[]> _pGrid;
+    _terminalChar_t defaultNullChar = { 0, 0, 0, false };
+    
     uint8_t _curX = 0;
     uint8_t _curY = 0;
-    std::unique_ptr<terminalChar_t[]> _pGrid;
-    terminalChar_t defaultNullChar = { 0, 0, 0, false };
     uint8_t _viewPortX;
     uint8_t _viewPortWidth;
     uint8_t _viewPortY;
     uint8_t _viewPortHeight;
-
-    bool _cursonOn = true;
-    SDL_TimerID _cursorTimer = 0;
-    
-    // these should be parameters?
-    uint8_t cur_shape = 219;
-    uint32_t cursor_time = 500;
- 
-    uint32_t _timerId = 0;
+  
+    bool _drawCursor = true; 
+    SDL_TimerID _cursorTimerId = 0;
     static uint32_t _timerCallBack(uint32_t interval, void* param);
 
-    void incrementCursorPosition() noexcept;
-    void scrollDownGrid() noexcept;
-    void renderChar(const SDL_Point& dst, const uint8_t col, const uint8_t bgCol, const char c);
+    void _incrementCursorPosition(bool increment = true) noexcept;
+    void _scrollDownGrid() noexcept;
+    void _renderFontChar(const SDL_Point& dst, _terminalChar_t& tc);
+    void _renderCharLine(const std::bitset<8> line, const int dstx, const int dsty, const uint8_t col, const uint8_t bgCol);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Function parameter 'line' should be passed by const reference.](https://app.codacy.com/manual/Raffaello/sdl2-vga-terminal/pullRequest?prid=5403184)